frontend/src/metabase/visualizations/visualizations/SmartScalar.jsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import { Box, Flex } from "grid-styled";
+import { t, jt } from "c-3po";
+import _ from "underscore";
+
+import { formatNumber, formatValue } from "metabase/lib/formatting";
+import colors from "metabase/lib/colors";
+
+import Icon from "metabase/components/Icon";
+
+import * as Query from "metabase/lib/query/query";
+import * as Card from "metabase/meta/Card";
+import { parseFieldBucketing, formatBucketing } from "metabase/lib/query_time";
+
+import { columnSettings } from "metabase/visualizations/lib/settings/column";
+import {
+  MinRowsError,
+  NoBreakoutError,
+} from "metabase/visualizations/lib/errors";
+
+import ScalarValue, {
+  ScalarWrapper,
+  ScalarTitle,
+} from "metabase/visualizations/components/ScalarValue";
+
+export default class Smart extends React.Component {
+  static uiName = "Smart Scalar";
+  static identifier = "smartscalar";
+  static iconName = "smartscalar";
+
+  static minSize = { width: 3, height: 3 };
+
+  static noHeader = true;
+
+  _scalar: ?HTMLElement;
+
+  static settings = {
+    ...columnSettings({
+      getColumns: ([{ data: { cols } }], settings) => [
+        _.find(cols, col => col.name === settings["scalar.field"]) || cols[1],
+      ],
+    }),
+    "scalar.switch_positive_negative": {
+      title: t`Switch positive / negative colors?`,
+      widget: "toggle",
+    },
+  };
+
+  // Always return true here so that we don't discourage this feature.
+  static isSensible() {
+    return true;
+  }
+
+  // Smart scalars need to have a breakout
+  static checkRenderable(series, settings) {
+    const singleSeriesHasNoRows = ({ data: { cols, rows } }) => rows.length < 1;
+    if (_.every(series, singleSeriesHasNoRows)) {
+      throw new MinRowsError(1, 0);
+    }
+
+    const cols = series[0].data.cols;
+
+    if (cols.length < 2) {
+      throw new NoBreakoutError(
+        t`Group by a time field to see how this has changed over time`,
+      );
+    }
+  }
+
+  _getColumnIndex(cols: Column[], settings: VisualizationSettings) {
+    const columnIndex = _.findIndex(
+      cols,
+      col => col.name === settings["scalar.field"],
+    );
+    return columnIndex < 0 ? 1 : columnIndex;
+  }
+
+  render() {
+    const insights =
+      this.props.rawSeries &&
+      this.props.rawSeries[0].data &&
+      this.props.rawSeries[0].data.insights;
+    const {
+      onChangeCardAndRun,
+      onVisualizationClick,
+      isDashboard,
+      isFullscreen,
+      settings,
+      visualizationIsClickable,
+      series: [{ card, data: { rows, cols } }],
+    } = this.props;
+
```

Comment:
If you implement the `checkRenderable` I described above that should be sufficient, but it's probably a good idea to make sure `insights` isn't null before using it just in case. 

```
if (!insights) {
  return null
}
```

should be sufficient